### PR TITLE
[SourceKit] [NFC] Rename DEPENDS to LINK_LIBRARIES throughout SourceKit

### PR DESCRIPTION
**Pull request by harlanhaskins**
Currently, SourceKit's CMake functions all use DEPENDS to specify
libraries the targets will link with. This is confusing as it doesn't
behave the same way that add_swift behaves, and implies that
dependencies are created when there aren't.

This is the "proper fix" for #11226.

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by gottesmm** [comment]
@llvm-beanz Your thoughts?

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Review by benlangmuir — APPROVED** [review]

**Review by nkcsgexi — APPROVED** [review]

**Comment by llvm-beanz** [comment]
The general convention in LLVM is:

- `LINK_COMPONENTS` for LLVM Components
- `LINK_LIBS` for link dependencies that are not LLVM components
- `DEPENDS` for order dependencies

While I'm completely in favor of renaming `DEPENDS` because its use is confusing. I'd rather rename to `LINK_LIBS` rather than `LINK_LIBRARIES` so that we're consistent with all the other LLVM projects.

**Comment by harlanhaskins** [comment]
The rest of Swift uses `LLVM_COMPONENT_DEPENDS` instead of `LINK_COMPONENTS`, and I think it would be nice to explicitly namespace these parameters to `LLVM_LINK_COMPONENTS` within the Swift project.

I'd rather this patch not touch the rest of the Swift CMake because it's a very large surface area and I wanted to keep this in line with the existing conventions in the rest of Swift. This is something that I think should be standardized across the projects though.

That said, I'm more than happy to rename `LINK_LIBRARIES` to `LINK_LIBS` for this patch 👍

Thanks for taking a look!

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@llvm-beanz Good to merge once CI passes?

**Comment by gottesmm** [comment]
@harlanhaskins Can you rebase the branch before you merge?

**Comment by harlanhaskins** [comment]
@gottesmm I wish GitHub would rebase for its web conflicts resolver 😂 (I'm rebasing now)

**Comment by gottesmm** [comment]
@harlanhaskins I agree.

**Comment by harlanhaskins** [comment]
I did a bad find&replace so I need to manually fix this. Give me just a bit.

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
:boat:

**Pull request merged**
